### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by padenot**
We need to be able to ask people to enable logging and have a lot of info from machines we don't control, so I added a bunch of logging statements. Sometimes, the logging statement is only there to see where the program errored.

This includes logging about device enumeration, that I copied and adapted from `test_device.c`.

@kinetiknz, @achronop, can you have a look ?

### CODE REVIEW

File: src/cubeb_wasapi.cpp [hunk 1]
```diff
@@ -749,11 +756,16 @@ refill_callback_output(cubeb_stream * stm)
     return true;
   }
 
+
```
Comment (achronop) [inline src/cubeb_wasapi.cpp]: Nit: Extra new line

Comment (padenot) [inline src/cubeb_wasapi.cpp]: Fixed.

File: src/cubeb.c [hunk 2]
```diff
@@ -384,18 +384,113 @@ int cubeb_stream_register_device_changed_callback(cubeb_stream * stream,
   return stream->context->ops->stream_register_device_changed_callback(stream, device_changed_callback);
 }
 
+void log_device(cubeb_device_info * device_info)
```
Comment (kinetiknz) [inline src/cubeb.c]: Should be static

File: src/cubeb_audiounit.cpp [hunk 3]
```diff
@@ -1180,6 +1180,7 @@ setup_audiounit_stream(cubeb_stream * stm)
   UInt32 size;
 
   if (has_input(stm)) {
+    LOG("Opening input side, rate: %d", stm->input_stream_params.rate);
```
Comment (kinetiknz) [inline src/cubeb_audiounit.cpp]: Should be %u, rate is unsigned

File: src/cubeb_audiounit.cpp [hunk 4]
```diff
@@ -1190,6 +1191,7 @@ setup_audiounit_stream(cubeb_stream * stm)
   }
 
   if (has_output(stm)) {
+    LOG("Opening output side, rate: %d", stm->input_stream_params.rate);
```
Comment (kinetiknz) [inline src/cubeb_audiounit.cpp]: Ditto

**Pull request merged**